synapse/module_api/__init__.py

```diff
@@ -279,7 +279,11 @@ def complete_sso_login(
         )
 
     async def complete_sso_login_async(
-        self, registered_user_id: str, request: SynapseRequest, client_redirect_url: str
+        self,
+        registered_user_id: str,
+        request: SynapseRequest,
+        client_redirect_url: str,
+        new_user: bool = False,
```

Comment:
Does it make sense to add this to the `complete_sso_login` version as well? I'm not sure what even calls these anymore.

Reply:
given it's deprecated, I'm not sure there's any value in updating it. Any modules which want to modify the value of `new_user` should call the async version.